### ISSUE TITLE
Gretty 1.2.4 groovy classloader issue on tomcat only

### ISSUE THREAD

**Issue by kpage** [akhikhl/gretty#191]
Hi @akhikhl !  Thanks so much for your work on gretty, it has helped me immensely.

I noticed an issue with the groovy classloader after upgrading to gretty 1.2.4.  This issue is not present on gretty 1.2.2 (My build was not able to find gretty 1.2.3 in the repo, so 1.2.2 was the last working version).

My app uses liquibase with a groovy configuration file.  It seems like gretty is interfering with the groovy classloader used inside the web app.  This only happens on Tomcat (works fine on jetty) and only if Tomcat was loaded within gretty.  Deploying the WAR to a manually-installed Tomcat 8 works fine.

Liquibase is configured to run during server startup as a Spring bean.  The actual exception that happens when liquibase runs is:

```
java.lang.ClassCastException: java.lang.String cannot be cast to java.io.File
	at org.codehaus.groovy.runtime.dgm$1011.invoke(Unknown Source) ~[groovy-all-2.3.10.jar:2.3.10]
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite$PojoMetaMethodSiteNoUnwrapNoCoerce.invoke(PojoMetaMethodSite.java:271) ~[groovy-all-2.3.10.jar:2.3.10]
	at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:53) ~[groovy-all-2.3.10.jar:2.3.10]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:45) ~[groovy-all-2.3.10.jar:2.3.10]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:110) ~[groovy-all-2.3.10.jar:2.3.10]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:122) ~[groovy-all-2.3.10.jar:2.3.10]
	at liquibase.parser.ext.GroovyLiquibaseChangeLogParser.parse(GroovyLiquibaseChangeLogParser.groovy:45) ~[liquibase-groovy-dsl-1.1.0.jar:na]
	at liquibase.Liquibase.getDatabaseChangeLog(Liquibase.java:215) ~[liquibase-core-3.3.3.jar:na]
	at liquibase.Liquibase.update(Liquibase.java:192) ~[liquibase-core-3.3.3.jar:na]
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:402) ~[liquibase-core-3.3.3.jar:na]
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:366) ~[liquibase-core-3.3.3.jar:na]
```

As you can see in that stack trace, groovy-all-2.3.10 is being used.  liquibase-groovy-dsl declares a dependency on groovy 2.4.1 and thus should be using that version of groovy.  groovy-all-2.3.10 is being imported by gretty.

```
+--- org.liquibase:liquibase-groovy-dsl:1.1.0
|    +--- org.codehaus.groovy:groovy-sql:2.4.1
|    |    \--- org.codehaus.groovy:groovy:2.4.1
|    +--- org.liquibase:liquibase-core:3.3.2 -> 3.3.3
|    \--- org.codehaus.groovy:groovy:2.4.1
```

Minimal apps reproducing the issue are listed below.  One uses Spring Boot and one uses Spring XML configuration:

https://github.com/kpage/gretty-groovy-classloader
https://github.com/kpage/gretty-groovy-classloader-xml

**Comment by matthew-b-payne** [comment]
Yup, previously this bug was present in jetty as well.

**Comment by johnjcun** [comment]
I can confirm that this went away when trying jetty9 but was there for tomcat 7 and 8. I was using 1.2.4.
I also tried with 1.2.2 but tomcat failed also. We will use jetty9 for testing for now but would really appreciate a fix for this on tomcat 7 - its worked great until we moved to Java 8.

**Comment by kpage** [comment]
This is fixed in gretty 1.4.0
